src/metabase-report-executor/src/main/java/com/champtitles/metabasereportexecutor/executor/MetabaseClient.java

```diff
@@ -0,0 +1,245 @@
+        String response = sendHttpRequest(httpRequest, 200);
+
+        try {
+            return objectMapper.readValue(response, SessionPropertiesResponse.class);
+        } catch (JsonProcessingException e) {
+            logger.error("failed to parse session properties response: {}", response);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Complete the initial setup of the metabase server by creating a user and setting basic properties
+     * <p>
+     * metabase.com/docs/latest/api/setup#post-apisetup
+     *
+     * @param setupToken setup token obtained by querying the session properties API
+     */
+    public void completeInitialSetup(String setupToken) {
+        SetupRequest setupRequest = new SetupRequest(setupToken, email, password);
+
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/setup"))
+                .header("Content-Type", "application/json")
+                .POST(createBody(setupRequest))
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+        logger.info("setup response: {}", response);
+    }
+
+    /**
+     * Log in to metabase with the configured email and password and store the session id
+     * <p>
+     * metabase.com/docs/latest/api/session#post-apisession
+     *
+     * @return session id (cookie) to be used for subsequent requests
+     */
+    public String loginAndGetSession() {
+        SessionRequest sessionRequest = new SessionRequest(email, password);
+
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/session"))
+                .header("Content-Type", "application/json")
+                .POST(createBody(sessionRequest))
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+
+        try {
+            SessionResponse sessionResponse = objectMapper.readValue(response, SessionResponse.class);
+            logger.info("logged in successfully");
+            logger.debug("session id: {}", sessionResponse.id());
+            sessionId = sessionResponse.id();
+            return sessionResponse.id();
+
+        } catch (JsonProcessingException e) {
+            logger.error("failed to parse session properties response: {}", response);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Create a Card in Metabase
+     * <p>
+     * metabase.com/glossary/card
+     * metabase.com/docs/latest/api/card#post-apicard
+     *
+     * @param name name of the card to create
+     * @return id of the created card
+     */
+    public String createCard(String name) {
+        if (StringUtils.isBlank(sessionId)) {
+            throw new RuntimeException("you must login before creating a card");
+        }
+
+        CreateCardRequest createCardRequest = new CreateCardRequest(name);
+
+        HttpRequest httpRequest = HttpRequest
```

Comment:
This could be wrapped into a method that provides some convenience, but low value currently.